src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -168,6 +169,32 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
         m_subrequests_gio.at(sub_idx).global_results[i] = out_idx;
     }
     // }}}
+
+    // Sort out how to handle weights bank and unpack
+    for (size_t i = 0; i < m_num_submodels; i++) {
+        auto& comp_model_desc = m_npuw_model->m_compiled_submodels[i];
+
+        if (!comp_model_desc.compiled_model && !comp_model_desc.replaced_by) {
+            continue;
+        }
+
+        auto& request = m_subrequests[i];
+
+        for (std::size_t cidx = 0u; cidx < comp_model_desc.closure.size(); cidx++) {
+            const auto& closure = comp_model_desc.closure[cidx];
+
+            const auto closure_param_id = comp_model_desc.param_base + cidx;
+            const auto& iport = comp_model_desc.compiled_model->inputs()[closure_param_id];
+            const auto& clparam = request->get_tensor(iport);
+
+            if (closure.get_element_type() == clparam->get_element_type()) {
+                comp_model_desc.unpack_required[cidx] = false;
+            } else {
+                comp_model_desc.unpack_required[cidx] = true;
+            }
```

Comment:
`unpack_required` could be sorted in the same place where you initialize `update_required` -- in the CompiledModel

Reply:
Done